src/main/java/org/broadinstitute/hellbender/utils/Log10FactorialCache.java

```diff
@@ -0,0 +1,20 @@
+package org.broadinstitute.hellbender.utils;
+
+import org.apache.commons.math3.special.Gamma;
+
+/**
+ * Wrapper class so that the log10Factorial array is only calculated if it's used
+ */
+public class Log10FactorialCache extends IntToDoubleFunctionCache {
+    private static final int CACHE_SIZE = 10_000;
+
+    @Override
+    protected int maxSize() {
+        return CACHE_SIZE;
+    }
+
+    @Override
+    protected double compute(final int n) {
+        return MathUtils.log10Gamma(n + 1);
```

Comment:
Without delving into the details of what exactly your branch is computing as of right now, it looks like this isn't actually doing the factorial computation here but rather calling out to the logGamma function. This may be equivalent? If not then you should update the comments on this class to reflect what its doing.

Reply:
Yes, the gamma function satisfied Gamma(n + 1) = n! for integers n.  The old implementation instead grew the cache incrementally as log((n+1)!) = log(n!) + log(n+1).  Since log(n+1) in the old implementation was also cached this was faster, but the one-time cost of computing logGamma a few thousand times to fill the cache is probably a few milliseconds, if that.